pkg/api/http_server.go

```diff
@@ -143,25 +143,16 @@ func (hs *HTTPServer) listenAndServeTLS(certfile, keyfile string) error {
 
 	tlsCfg := &tls.Config{
 		MinVersion:               tls.VersionTLS12,
-		PreferServerCipherSuites: true,
+		PreferServerCipherSuites: false, // we support only safe ciphers, so client may decide preferred cipher
 		CipherSuites: []uint16{
-			tls.TLS_RSA_WITH_AES_128_CBC_SHA,
-			tls.TLS_RSA_WITH_AES_256_CBC_SHA,
-			tls.TLS_RSA_WITH_AES_128_GCM_SHA256,
-			tls.TLS_RSA_WITH_AES_256_GCM_SHA384,
-			tls.TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA,
-			tls.TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA,
-			tls.TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA,
-			tls.TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA,
-			tls.TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,
-			tls.TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,
 			tls.TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,
-			tls.TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,
+			tls.TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305,  // the best cipher for mobile devices without HW AES support - Snapdragon 800-
+			tls.TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256, // must be enabled for http/2 support
```

Comment:
Can you explain the changes here why the change to PreferServerCipherSuites, and the removal of cipher, what implications does this change have?

Reply:
Some weak ciphers are on the h2 blacklist - https://github.com/golang/go/blob/master/src/net/http/h2_bundle.go#L397-L686, so they can't be used.

PreferServerCipherSuites - there is no official recommendation for client/server cipher preference. Generally, if support for weak ciphers is required, then server ciphers should be preferred. If you have a support only for secure ciphers, then the client can decide which cipher is the best one. The best blog post for cipher problem: https://hynek.me/articles/hardening-your-web-servers-ssl-ciphers/

This is WIP and it needs a discussion - for example list and order of supported ciphers.